### DISCUSSION TITLE
Missing Libraries

### DISCUSSION THREAD

**Discussion by sdcycling** [ImageMagick/ImageMagick#4845] [Help]
Hi, I am trying to install imagemagick on a MacBook Air (M1,2020) running MacOS Monterey 12.1.  I tried the following test:

% magick logo: logo.gif

A library does not load:

dyld[9508]: Library not loaded: /ImageMagick-7.0.10/lib/libMagickCore-7.Q16HDRI.8.dylib
  Referenced from: /Users/douglasdommermuth/ImageMagick-7.0.10/bin/magick
  Reason: tried: '/ImageMagick-7.0.10/lib/libMagickCore-7.Q16HDRI.8.dylib' (no such file), '/usr/local/lib/libMagickCore-7.Q16HDRI.8.dylib' (no such file), '/usr/lib/libMagickCore-7.Q16HDRI.8.dylib' (no such file)
Abort

It seems to be looking in the right area, but here are the files in that directory:

% ls lib
ImageMagick-7.0.10		libMagick++-7.Q16HDRI.la	libMagickCore-7.Q16HDRI.la	libMagickWand-7.Q16HDRI.la
libMagick++-7.Q16HDRI.4.dylib	libMagickCore-7.Q16HDRI.8.dylib	libMagickWand-7.Q16HDRI.8.dylib	pkgconfig
libMagick++-7.Q16HDRI.a		libMagickCore-7.Q16HDRI.a	libMagickWand-7.Q16HDRI.a
libMagick++-7.Q16HDRI.dylib	libMagickCore-7.Q16HDRI.dylib	libMagickWand-7.Q16HDRI.dylib

What am I doing wrong?   Thank you, Doug.

**Comment by fmw42** [comment]
What install did you get?  Is it a binary or source code only?  Perhaps you have no delegates installed.

**Reply by fmw42** [reply]
Recommend installing from Homebrew binary for Imagemagick 7.

**Reply by sdcycling** [reply]
Hi,  I got ImageMagick-x86_64-apple-darwin20.1.0.tar.gz on https://imagemagick.org/script/download.php.  Thank you.

**Reply by sdcycling** [reply]
I will try installing Homebrew.

**Reply by fmw42** [reply]
Read that page carefully. That binary is only for macOS High Sierra.

**Reply by sdcycling** [reply]
Homebrew worked.   I had to make sure that the path for brew was set for Apple Silicon.   Since I had used Apple Time machine to migrate from an Apple Intel mac to an Apple Silicon mac, brew was pointing to /usr/local/bin instead of /opt/homebrew/bin.  Once I set it to the correct path for Apple Silicon, the install of Imagemagick worked.
